src/Wangkanai.Detection/DetectionService.cs

```
@@ -18,18 +18,17 @@ public class DetectionService : IDetectionService
 
         public DetectionService(IServiceProvider services)
         {
-            if (services != null) Context = services.GetService<IHttpContextAccessor>()?.HttpContext;
-
-            UserAgent = CreateUserAgent(Context);
+            if (services == null) throw new ArgumentNullException(nameof(services));
```

Fail fast, return fast